src/app/frontend/common/components/date/component.ts

```diff
@@ -36,14 +38,90 @@ import {Component, Input} from '@angular/core';
   selector: 'kd-date',
   templateUrl: './template.html',
   styleUrls: ['./style.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DateComponent {
+export class DateComponent implements OnChanges, OnDestroy {
   @Input() date: string;
   @Input() format = 'medium';
 
-  _relative: boolean;
   @Input('relative')
   set relative(v: boolean) {
-    this._relative = v !== undefined && v !== false;
+    this.relative_ = v !== undefined && v !== false;
+  }
+
+  get relative(): boolean {
+    return this.relative_;
+  }
+
+  iteration = 0;
+
+  private relative_: boolean;
+  private refreshInterval_: number;
+  private intervalChanged_ = new Subject<void>();
+  private timeBaseIntervals_ = [
+    60, // Seconds in a minute
+    60, // Minutes in a second
+    24, // Hours in a day
+  ];
+  private unsubscribe_ = new Subject<void>();
+
+  constructor(private readonly cdr_: ChangeDetectorRef) {}
+
+  ngOnChanges() {
```

Comment:
@Eddman I have updated the logic to what I have had in mind. I wanted to calculate the refresh time based on time that has already passed. This way refresh time is adjusted separately for every item on the list.

Reply:
Had a similar thought, but did not want to play around with the `this.date` 🙂 Anyway LGTM 😉 👍

Reply:
As a precaution, we could probably transform both dates to a timezone local to the browser and then do the calculation as the server might use a different timezone than the browser.

Reply:
I'm not following now, `new Date(this.date)` parses using ISO format and the `.getTime()` returns timestamp in UTC, or do I miss something?

Reply:
So we're comparing 2 timestamps in UTC I think...

Reply:
Just did a quick test in console to verify and it is UTC timestamp 🙂

Reply:
To be extra careful I have updated it to first convert time to the UTC and then calculate the difference.

Reply:
That I guess is overkill :) you can try that in console try setting offset on a date and you’ll see that getTime is always the same ;)

Reply:
True, pod `creationTimestamp` according to the documentation is in UTC also. I guess we could assume that `new Date().getTime()` will always return local time in UTC format. Both of them are UTC then.